src/wazuh_modules/wm_office365.c

```diff
@@ -33,24 +40,112 @@ void * wm_office365_main(wm_office365* office365_config) {
+{
+    wm_office365_auth* current = office365_auth;
+    wm_office365_auth* next = NULL;
+    while (current != NULL)
+    {
+        next = current->next;
+        os_free(current->tenant_id);
+        os_free(current->client_id);
+        os_free(current->client_secret_path);
+        os_free(current->client_secret);
+        os_free(current);
+        current = next;
+    }
+    office365_auth = NULL;
+}
+
+cJSON *wm_office365_dump(const wm_office365* office365_config) {
     cJSON *root = cJSON_CreateObject();
     cJSON *wm_info = cJSON_CreateObject();
 
     if (office365_config->enabled) {
-        cJSON_AddStringToObject(wm_info,"enabled","yes");
+        cJSON_AddStringToObject(wm_info, "enabled", "yes");
+    } else {
+        cJSON_AddStringToObject(wm_info, "enabled", "no");
+    }
+    if (office365_config->run_on_start) {
+        cJSON_AddStringToObject(wm_info, "run_on_start", "yes");
+    } else {
+        cJSON_AddStringToObject(wm_info, "run_on_start", "no");
+    }
+    if (office365_config->skip_on_error) {
+        cJSON_AddStringToObject(wm_info, "skip_on_error", "yes");
+    } else {
+        cJSON_AddStringToObject(wm_info, "skip_on_error", "no");
+    }
+    if (office365_config->only_future_events) {
+        cJSON_AddStringToObject(wm_info, "only_future_events", "yes");
     } else {
-        cJSON_AddStringToObject(wm_info,"enabled","no");
+        cJSON_AddStringToObject(wm_info, "only_future_events", "no");
     }
-    cJSON_AddItemToObject(root,"office365",wm_info);
+    if (office365_config->interval) {
+        cJSON_AddNumberToObject(wm_info, "interval", office365_config->interval);
+    }
+    if (office365_config->auth) {
+        wm_office365_auth *iter;
+        cJSON *arr_auth = cJSON_CreateArray();
+        for (iter = office365_config->auth; iter; iter = iter->next) {
+            cJSON *api_auth = cJSON_CreateObject();
+            if (iter->tenant_id) {
+                cJSON_AddStringToObject(api_auth, "tenant_id", iter->tenant_id);
+            }
+            if (iter->client_id) {
+                cJSON_AddStringToObject(api_auth, "client_id", iter->client_id);
+            }
+            if (iter->client_secret_path) {
+                cJSON_AddStringToObject(api_auth, "client_secret_path", iter->client_secret_path);
+            }
+            if (iter->client_secret) {
+                cJSON_AddStringToObject(api_auth, "client_secret", iter->client_secret);
+            }
+            cJSON_AddItemToArray(arr_auth, api_auth);
+        }
+        if (cJSON_GetArraySize(arr_auth) > 0) {
+            cJSON_AddItemToObject(wm_info, "api_auth", arr_auth);
+        } else {
+            cJSON_free(arr_auth);
+        }
+    }
+    cJSON *arr_subscription = cJSON_CreateArray();
+
+    cJSON *api_subscription = cJSON_CreateObject();
+    if (office365_config->subscription.azure) {
+        cJSON_AddStringToObject(api_subscription, "subscription", "Audit.AzureActiveDirectory");
```

Comment:
Is it allowed to add several items with the same key to a single object? Why don't you just add the subscriptions names to the subscriptions array?